### PR TITLE
Cypress testing

### PR DESCRIPTION
**Pull request by NguyenDa18**
Added Cypress E2E tests for easy validation and data input of application (e.g. tests different error messages, automated input to form fields)
This way the application is immediately interactive with newcomers and helps you cover every feature the application has.
[Video Demo](https://drive.google.com/file/d/1z7UR7ApbHi5_DjuFkUUEUbBQE9RlsOJV/view)

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,14 +3,14 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "axios": "^0.18.0",
+    "axios": "^0.18.1",
```
Comment (Omar-Handouk) [inline client/package.json]: Already addressed in pull #59

File: client/package.json [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
     "moment": "^2.24.0",
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
     "react-moment": "^0.9.2",
     "react-redux": "^7.0.2",
     "react-router-dom": "^5.0.0",
-    "react-scripts": "3.0.0",
+    "react-scripts": "^3.3.0",
```
Comment (Omar-Handouk) [inline client/package.json]: Already addressed in pull #59

File: cypress/integration/devconnector/test-create-profile.spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
+const baseURL = `http://localhost:3000`
```
Comment (Omar-Handouk) [inline cypress/integration/devconnector/test-create-profile.spec.js]: baseURL's port needs to be changed as a configuration variable or env var
```js
const baseURL = `http://localhost:${process.env.PORT}`
```

File: client/package.json (hunk 2 above)
Comment (Omar-Handouk) [inline client/package.json]: Already addressed in pull #59

File: cypress/integration/devconnector/test-login.spec.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const baseURL = `http://localhost:3000`
```
Comment (Omar-Handouk) [inline cypress/integration/devconnector/test-login.spec.js]: baseURL's port needs to be changed as a configuration variable or env var
```js
const baseURL = `http://localhost:${process.env.PORT}`
```

File: cypress/integration/devconnector/test-register.spec.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const baseURL = `http://localhost:3000`
```
Comment (Omar-Handouk) [inline cypress/integration/devconnector/test-register.spec.js]: baseURL's port needs to be changed as a configuration variable or env var
```js
const baseURL = `http://localhost:${process.env.PORT}`
```

**Comment by bushblade** [comment]
As @Omar-Handouk has said, the course doesn't cover testing and given that conflicts would arise from merging this feature, I'll close.

**Pull request closed without merging**